rendering/mmd_pmx.ksy

```diff
@@ -0,0 +1,569 @@
+  - id: bone_count
+    type: u4
+  - id: bones
+    type: bone
+    repeat: expr
+    repeat-expr: bone_count
+  - id: morph_count
+    type: u4
+  - id: morphs
+    type: morph
+    repeat: expr
+    repeat-expr: morph_count
+  - id: frame_count
+    type: u4
+  - id: frames
+    type: frame
+    repeat: expr
+    repeat-expr: frame_count
+  - id: rigid_body_count
+    type: u4
+  - id: rigid_bodies
+    type: rigid_body
+    repeat: expr
+    repeat-expr: rigid_body_count
+  - id: constraint_count
+    type: u4
+  - id: constraints
+    type: constraint
+    repeat: expr
+    repeat-expr: constraint_count
+
+types:
+
+  header:
+    seq:
+      - id: magic
+        contents: 'PMX '
+      - id: version
+        type: f4
+      - id: header_size
+        type: u1
+      - id: encoding
+        type: u1
+      - id: additional_uv_count
+        type: u1
+      - id: vertex_index_size
+        type: u1
+      - id: texture_index_size
+        type: u1
+      - id: material_index_size
+        type: u1
+      - id: bone_index_size
+        type: u1
+      - id: morph_index_size
+        type: u1
+      - id: rigid_body_index_size
+        type: u1
+      - id: model_name
+        type: len_string
+      - id: english_model_name
+        type: len_string
+      - id: comment
+        type: len_string
+      - id: english_comment
+        type: len_string
+
+  vertex:
+    seq:
+      - id: position
+        type: f4_3
+      - id: normal
+        type: f4_3
+      - id: uv
+        type: f4_2
+      - id: additional_uvs
+        type: f4_4
+        repeat: expr
+        repeat-expr: _root.header.additional_uv_count
+      - id: type
+        type: u1
+      - id: skin_indices
+        type:
+          switch-on: _root.header.bone_index_size
+          cases:
+            1: u1
+            2: u2
+            4: u4
+        repeat: expr
+        repeat-expr: 'type == 0 ? 1 : type == 1 ? 2 : type == 2 ? 4 : type == 3 ? 2 : -1'
+      - id: skin_weights
+        type: f4
+        if: type == 1 or type == 2 or type == 3
+        repeat: expr
+        repeat-expr: 'type == 1 ? 1 : type == 2 ? 4 : type == 3 ? 1 : -1'
+      - id: skin_c
+        type: f4_3
+        if: type == 3
+      - id: skin_r0
+        type: f4_3
+        if: type == 3
+      - id: skin_r1
+        type: f4_3
+        if: type == 3
+      - id: edge_ratio
+        type: f4
+
+  face:
+    seq:
+      - id: indices
+        type:
+          switch-on: _root.header.vertex_index_size
+          cases:
+            1: u1
+            2: u2
+            4: u4
+        repeat: expr
+        repeat-expr: 3
+
+  texture:
+    seq:
+      - id: name
+        type: len_string
+
+  material:
+    seq:
+      - id: name
+        type: len_string
+      - id: english_name
+        type: len_string
+      - id: diffuse
+        type: f4_4
+      - id: specular
+        type: f4_3
+      - id: shininess
+        type: f4
+      - id: ambient
+        type: f4_3
+      - id: flag
```

Comment:
Very ambigous name. Is it a single boolean (in this case name should start from `is_` and describe the meaning). Or is it binary flags (in this case it should be not `u1`, but a type, in which bit-sized numbers reside, `b1` is a bit and has boolean type instead)?

Reply:
Similar problem with the ones below as well.

I was able to figure out that this one contains _multiple_ flags, but have no docs on what the flags are. At least for now I'll rename it to `flags`. The ones below are more interesting though.